source/common/config/grpc_mux_impl.h

```diff
@@ -159,7 +160,10 @@ class NullGrpcMuxImpl : public GrpcMux,
 
   GrpcMuxWatchPtr addWatch(const std::string&, const std::set<std::string>&, SubscriptionCallbacks&,
                            OpaqueResourceDecoder&) override {
-    throw EnvoyException("ADS must be configured to support an ADS config source");
+    ExceptionUtil::throwEnvoyException("ADS must be configured to support an ADS config source");
+    // it should never reach here but the compiler complains and the alternative will be to change
+    // the return type of the parent virtual function to absl::<optional<current_return_type>>
+    return nullptr;
```

Comment:
Use NOT_REACHED here. I think you may also be able to use some type of "never returns" annotation on `throwEnvoyException`

Reply:
used noreturn keyword on function signature